python/pdo/contract/replication.py

```diff
@@ -0,0 +1,214 @@
+# Copyright 2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import sys
+import concurrent.futures
+import queue
+import threading
+
+from pdo.contract.state import ContractState
+from pdo.common.utility import are_the_urls_same
+
+import logging
+logger = logging.getLogger(__name__)
+
+class ReplicationException(Exception) :
+    """
+    A class to capture replication exceptions
+    """
+    pass
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+
+class Replicator(object):
```

Comment:
much easier to use a class if we are dealing with multiple contracts, since the replication params are different for different contracts. even for single contract, it is easier to create an instance for each replication, since the blocks to replicate change.  

i agree there are many class variables and functions, i just find it convenient to attach a replicator to every response object that replicates just the specific change set. No different than the use of classes for request or responses. one per update.

Reply:
so then make the class JUST the replication parameters.